static/js/presence.js

```diff
@@ -40,7 +41,12 @@ export function is_active(user_id) {
 
 export function get_status(user_id) {
     if (people.is_my_user_id(user_id)) {
-        return "active";
+        if (user_settings.presence_enabled === undefined || user_settings.presence_enabled) {
+            // if the current user is sharing presence, they always see themselves as online.
+            return "active";
+        }
+        // if the current user is not sharing presence, they always see themselves as offline.
+        return "offline";
```

Comment:
@alya question: Do we want this to display yourself as "orange" if you're idle?  (i.e. cases like your Zulip window is on a second monitor and you're not using it)?  I think `activity.compute_active_status()` is the function that indicates what we send to the server.  

While this would be more accurate, the main downside is that you might see an immediate shift from orange=>green in the sidebar when alt-tabbing to a Zulip window.  (At least if we correctly had `mark_client_active` call the redraw function).

Reply:
We should try making it accurate and test the UX on CZO.

Reply:
OK; merged this since it fixes a much more visible problem, but made it not close the issue, since this issue remains open.

Reply:
#18846 shows as closed for me